File: exercise/java/day09/src/test/java/gift/SantaServiceTests.java
package gift;

import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;

class SantaServiceTests {

    private final SantaService service = new SantaService();

    @Test
    void requestIsApprovedForNiceChildWithFeasibleGift() {
        Child niceChild = new Child("Alice", "Thomas", 9, Behavior.NICE, new GiftRequest("Bicycle", true, Priority.NICE_TO_HAVE));
        assertThat(service.evaluateRequest(niceChild)).isTrue();
    }

    @Test
    void requestIsDeniedForNaughtyChild() {
        Child naughtyChild = new Child("Noa", "Thierry", 6, Behavior.NAUGHTY, new GiftRequest("SomeToy", true, Priority.DREAM));
        assertThat(service.evaluateRequest(naughtyChild)).isFalse();
    }

    @Test
    void requestIsDeniedForNiceChildWithInfeasibleGift() {
        Child niceChildWithInfeasibleGift = new Child("Charlie", "Joie", 3, Behavior.NICE, new GiftRequest("AnotherToy", false, Priority.DREAM));
        assertThat(service.evaluateRequest(niceChildWithInfeasibleGift)).isFalse();
    }
}

File: exercise/C#/day09/GiftWish.Tests/SantaServiceTests.cs
using FluentAssertions;
using Xunit;

namespace GiftWish.Tests
{
    public class SantaServiceTests
    {
        private readonly SantaService _service = new();

        [Fact]
        public void RequestIsApprovedForNiceChildWithFeasibleGift()
        {
            var niceChild = new Child("Alice", "Thomas", 9, Behavior.Nice, new GiftRequest("Bicycle", true, Priority.NiceToHave));
            _service.EvaluateRequest(niceChild).Should().BeTrue();
        }

        [Fact]
        public void RequestIsDeniedForNaughtyChild()
        {
            var naughtyChild = new Child("Noa", "Thierry", 6, Behavior.Naughty, new GiftRequest("SomeToy", true, Priority.Dream));
            _service.EvaluateRequest(naughtyChild).Should().BeFalse();
        }

        [Fact]
        public void RequestIsDeniedForNiceChildWithInfeasibleGift()
        {
            var niceChildWithInfeasibleGift = new Child("Charlie", "Joie", 3, Behavior.Nice, new GiftRequest("AnotherToy", false, Priority.Dream));
            _service.EvaluateRequest(niceChildWithInfeasibleGift).Should().BeFalse();
        }
    }
}
